### DISCUSSION TITLE
[TypeScript] Usage LIFT with typescript makes strange import errors...

### DISCUSSION THREAD

**Discussion by Setitch** [getlift/lift#187] [General]
### Description

When i created queue with handler of TS file, and then deployed code, i get following error each time something goes into the queue

```
2022-04-19T10:35:25.209Z	undefined	ERROR	Uncaught Exception 	
{
    "errorType": "Runtime.ImportModuleError",
    "errorMessage": "Error: Cannot find module 'offboarding-started'\nRequire stack:\n- /var/runtime/UserFunction.js\n- /var/runtime/index.js",
    "stack": [
        "Runtime.ImportModuleError: Error: Cannot find module 'job-started'",
        "Require stack:",
        "- /var/runtime/UserFunction.js",
        "- /var/runtime/index.js",
        "    at _loadUserApp (/var/runtime/UserFunction.js:202:13)",
        "    at Object.module.exports.load (/var/runtime/UserFunction.js:242:17)",
        "    at Object.<anonymous> (/var/runtime/index.js:43:30)",
        "    at Module._compile (internal/modules/cjs/loader.js:1085:14)",
        "    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)",
        "    at Module.load (internal/modules/cjs/loader.js:950:32)",
        "    at Function.Module._load (internal/modules/cjs/loader.js:790:12)",
        "    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:75:12)",
        "    at internal/main/run_main_module.js:17:47"
    ]
}
```

If i change the file name from  job-started.handler.ts to job-started.ts  i get following error (not matter how i export the function - i used every export method there is).
```
2022-04-19T10:08:12.514Z	undefined	ERROR	Uncaught Exception 	
{
    "errorType": "Runtime.HandlerNotFound",
    "errorMessage": "src/queueHandlers/jobs/job-started-handler.ts is undefined or not exported",
    "stack": [
        "Runtime.HandlerNotFound: src/queueHandlers/jobs/job-started-handler.ts is undefined or not exported",
        "    at Object.module.exports.load (/var/runtime/UserFunction.js:246:11)",
        "    at Object.<anonymous> (/var/runtime/index.js:43:30)",
        "    at Module._compile (internal/modules/cjs/loader.js:1085:14)",
        "    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)",
        "    at Module.load (internal/modules/cjs/loader.js:950:32)",
        "    at Function.Module._load (internal/modules/cjs/loader.js:790:12)",
        "    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:75:12)",
        "    at internal/main/run_main_module.js:17:47"
    ]
}

### How to Reproduce

job-started:
  type: queue
  worker:
    handler: src/queueHandlers/jobs/job-started.handler.ts


### Additional Information

_No response_

**Comment by mnapoli** [comment]
Hi! I moved this to a discussion since this isn't a Lift bug.

The `handler` field is the same as any other function in `serverless.yml`. This shouldn't be a file name, but rather `<file name>.<function name>`.

So here maybe try:

```yaml
handler: src/queueHandlers/jobs/job-started.handler
```

or something of the sort?

**Reply by Setitch** [reply]
There is slight issue, basically i started from using this file naming convention:   handles/[module-name]/[file-name].handler.ts   and there is no way to use that in lift. (even if file is also exporting handler as identifier). So it would be great either to add - worker as file & handler, or current method - just worker with [file-name-with-many-restrictions].[handler function]

**Reply by mnapoli** [reply]
The `handler` property should work exactly like any other function in serverless.yml.

Let me know if there's any difference, in that case it would be a bug. But is the syntax you are using working for a non-Lift function in serverless.yml?

**Reply by Setitch** [reply]
i used   job-start.handler.handler    and i think it worked, but looked tooo stupid i must say, so i think (as dot is not allowed as we see) to use dashed instead. (the funny thing is that   sometimes all works sometimes do not, end the error messages are not always helping. Thanks for straightening how the `handler` here is composed. Would be great to add information about dot in file names in the docs of lift^_^
